tests/components/discord/test_config_flow.py

```diff
@@ -0,0 +1,206 @@
+"""Test Discord config flow."""
+import nextcord
+from pytest import LogCaptureFixture
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.discord.const import DOMAIN
+from homeassistant.const import CONF_API_TOKEN, CONF_SOURCE
+from homeassistant.core import HomeAssistant
+
+from . import (
+    CONF_DATA,
+    CONF_IMPORT_DATA,
+    CONF_IMPORT_DATA_NO_NAME,
+    CONF_INPUT,
+    NAME,
+    create_entry,
+    mock_exception,
+    mocked_discord_info,
+    patch_discord_login,
+)
+
+
+async def test_flow_user(hass: HomeAssistant) -> None:
+    """Test user initialized flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+    )
+    with mocked_discord_info(), patch_discord_login():
+        result = await hass.config_entries.flow.async_configure(
```

Comment:
We normally patch the integration setup function on all create entry and reauth_successful results to avoid setting up the whole integration.